src/zoo_mcp/zoo_tools.py

```diff
@@ -1077,42 +1126,32 @@ async def zoo_snapshot_of_kcl(
 
     logger.info("Taking a snapshot of KCL")
 
-    # default to using the code if both are provided
-    if kcl_code and kcl_path:
-        logger.warning("Both code and kcl_path provided, using code")
-        kcl_path = None
-
-    if kcl_path:
-        kcl_path = Path(kcl_path)
-        if kcl_path.is_file() and kcl_path.suffix != ".kcl":
-            logger.error("The provided kcl_path is not a .kcl file")
-            raise ZooMCPException("The provided kcl_path is not a .kcl file")
-        if kcl_path.is_dir() and not (kcl_path / "main.kcl").is_file():
-            logger.error(
-                "The provided kcl_path directory does not contain a main.kcl file"
-            )
-            raise ZooMCPException(
-                "The provided kcl_path does not contain a main.kcl file"
-            )
-
-    if not kcl_code and not kcl_path:
-        logger.error("Neither code nor kcl_path provided")
-        raise ZooMCPException("Neither code nor kcl_path provided")
+    kcl_code, kcl_path = _check_kcl_code_or_path(kcl_code, kcl_path)
 
     view = kcl.SnapshotOptions(camera=camera, padding=padding)
 
     if kcl_code:
-        jpeg_contents_list = await kcl.execute_code_and_snapshot_views(
-            kcl_code, kcl.ImageFormat.Jpeg, snapshot_options=[view]
+        # The stub says list[list[int]] but it actually returns list[bytes]
+        jpeg_contents_list: list[bytes] = cast(
+            list[bytes],
+            cast(
+                object,
+                await kcl.execute_code_and_snapshot_views(
+                    kcl_code, kcl.ImageFormat.Jpeg, snapshot_options=[view]
+                ),
+            ),
         )
     else:
         assert isinstance(kcl_path, Path)
-        jpeg_contents_list = await kcl.execute_and_snapshot_views(
-            str(kcl_path), kcl.ImageFormat.Jpeg, snapshot_options=[view]
+        # The stub says list[list[int]] but it actually returns list[bytes]
+        jpeg_contents_list = cast(
+            list[bytes],
+            cast(
+                object,
+                await kcl.execute_and_snapshot_views(
+                    str(kcl_path), kcl.ImageFormat.Jpeg, snapshot_options=[view]
+                ),
+            ),
         )
```

Comment:
Same comment as above, I think the inner `cast(object, ...)` should be unnecessary.